raft/src/test/java/org/apache/kafka/snapshot/SnapshotWriterReaderTest.java

```diff
@@ -49,7 +48,7 @@
 
 public final class SnapshotWriterReaderTest {
     private final int localId = 0;
-    private final Set<Integer> voters = Collections.singleton(localId);
+    private final Set<Integer> voters = Set.of(localId);
 
     @Test
     public void testSnapshotDelimiters() throws Exception {
```

Comment:
Could you also check the assertion where the expected position is incorrect in L224 and L244?